platform-tests/src/jmh/java/org/junit/jupiter/jmh/PlatformBenchmarks.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright 2015-2018 the original author or authors.
+ *
+ * All rights reserved. This program and the accompanying materials are
+ * made available under the terms of the Eclipse Public License v2.0 which
+ * accompanies this distribution and is available at
+ *
+ * http://www.eclipse.org/legal/epl-v20.html
+ */
+
+package org.junit.jupiter.jmh;
+
+import static org.junit.platform.engine.discovery.DiscoverySelectors.selectClass;
+import static org.junit.platform.launcher.core.LauncherConfig.builder;
+import static org.junit.platform.launcher.core.LauncherDiscoveryRequestBuilder.request;
+import static org.junit.platform.launcher.core.LauncherFactory.create;
+
+import org.junit.jupiter.engine.JupiterTestEngine;
+import org.junit.platform.engine.TestEngine;
+import org.junit.platform.launcher.LauncherDiscoveryRequest;
+import org.junit.platform.launcher.core.LauncherConfig;
+import org.junit.runner.JUnitCore;
+import org.junit.vintage.engine.VintageTestEngine;
+import org.openjdk.jmh.annotations.Benchmark;
+
+/**
+ * JMH benchmarks for platform.
+ *
+ * @since 5.4
+ */
+public class PlatformBenchmarks {
```

Comment:
I'd consider adding a few annotations setting default parameter values appropriate to these benchmarks so that one does not have to always specify them themselves (or rely on defaults), e.g.,
```
@OutputTimeUnit(SECONDS)
@BenchmarkMode(Throughput)
? @Warmup(iterations = …)
? @Measurement(iterations = …)
@Fork(1)
```